### ISSUE TITLE
Currency#quick_withdraw_limit

### ISSUE THREAD

**Issue by sramsden** [openware/peatio#692]
I have a Question, not an Issue.

In the `Currency` class I don't see any benefit of the method ..
```
  def quick_withdraw_limit
    self[:quick_withdraw_limit] ||= BigDecimal.new(super.to_s)
  end
```
.. given that ...
```
validates :quick_withdraw_limit, numericality: { greater_than_or_equal_to: 0 }
```
.. and ..
```
  create_table "currencies", force: :cascade do |t|
    ...
    t.decimal  "quick_withdraw_limit", precision: 32, scale: 16, default: 0.0, null: false
```

Can you explain it and give me an example that would result in failure if that method did not exist?

**Comment by yivo** [comment]
Hi

Pls check
https://github.com/rubykube/peatio/blob/master/app/models/withdraw.rb#L127
https://github.com/rubykube/peatio/blob/master/app/models/withdraw.rb#L135

**Comment by yivo** [comment]
Withdraws that exceed quick withdraw limit must be approved manually by admin.

**Comment by sramsden** [comment]
Sorry, you have misunderstood my question.
I understand the concept of a quick withdraw limit.

What I am asking is why does that method ..

https://github.com/rubykube/peatio/blob/c588290a77dedd9a8612ba371d3eac1e2da2242c/app/models/currency.rb#L68

.. need to exist? The `quick_withdraw_limit` is a defined attribute of the active model.

There is no need to define a getter method for it that does lazy intialization because it is already configured to have a default zero value and also validated to be numeric.

In fact, if it could be blank .. `BigDecimal.new(super.to_s)` .. would result in .. `BigDecimal.new('')` .. which would result in and error... `ArgumentError (invalid value for BigDecimal(): "")`

**Comment by yivo** [comment]
@sramsden This was probably done for comparisons with BitDecimal amounts. It can't be blank since this is required attribute.

**Comment by sramsden** [comment]
Right, so you agree that this method can be deleted from the codebase?

**Comment by yivo** [comment]
Closed via #721.
